dev/ese/published/inc/sync.hxx

```diff
@@ -1974,40 +1978,28 @@ class CSemaphoreState
 
         //    ctors / dtors
 
-        CSemaphoreState( const CSyncStateInitNull& null ) : m_cAvail( 0 ) {}
```

Comment:
CKernelSemaphorePool can go away with CManualResetSignal also moving to WaitOnAddress

Reply:
Yes, fully replacing the remaining uses of the kernel semaphore would be a nice thing, to my mind.

The actual scope seems to be a bit larger than just `CManualResetSignal`, as I think that we'd need to switch at least the `CAutoResetSignal`, `CManualResetSignal` and `CGate`.

When I was thinking about it previously, I came to a conclusion that it might be better to do that iteratively, perhaps in a series of separate patchsets. Which is why I have started with reimplementing just the `CSemaphore`.

What do you think?  Personally, I think that it might be worth trying to fully complete the work on the `CSemaphore` first and then to separately build up on it for the remaining primitives.

Reply:
I agree, it would increase the scope unnecessarily